pkg/xmtp/listener.go

```diff
@@ -185,12 +186,17 @@ func (l *Listener) processEnvelope(env *v1.Envelope) error {
 }
 
 func (l *Listener) shouldDeliver(messageContext interfaces.MessageContext, subscription interfaces.Subscription) bool {
-	if messageContext.ShouldPush != nil {
-		return *messageContext.ShouldPush
-	}
 	if subscription.HmacKey != nil && len(subscription.HmacKey.Key) > 0 {
 		isSender := messageContext.IsSender(subscription.HmacKey.Key)
-		return !isSender
+		l.logger.Info("Has HMAC key", zap.Bool("is_sender", isSender))
+		if (isSender) {
+			return !isSender
+		}
```

Comment:
If it is not the sender we should return here instead we should fall down into the next check for if the codec says it should be pushed.

Reply:
🤦